### DISCUSSION TITLE
Range of THB1

### DISCUSSION THREAD

**Discussion by sanschaussure** [pvvx/THB2#68] [General]
Dear all, 
I bought several THB1 and flashed them with the latest version. I also have some Xiaomi Mijia, which are working fine with pvvx's firmware for nearly a year. To receive the signals I already installed an OpenMQTTGateway and a esphome ble proxy. Both on a ESP32 Wroom devices. The longest distance between the receiver and one Xiaomi device is 10m inside a building and the receiving of the signal is pretty stable.

Now i setup the new THB1 devices and had to find out that I cannot place the devices more than in the next room (~4m). The receiver cannot get the signal any more. The transmitting power at the sensor is set to +5dB.

Do you have the same experiences or do I have bad devices?

Thank you for your answer.

sanschaussure

**Comment by pvvx** [comment]
Most Tuya devices have very poorly made antennas.

---

Plus:

If the ESP32 module is installed incorrectly - the antenna is not brought out from the second printed circuit board, then it has poor reception sensitivity and weak transmission.
Examples:
Incorrect module installation:
![image](https://github.com/user-attachments/assets/19466bca-29f6-4379-88e0-c902887f6937)

Correct:
![image](https://github.com/user-attachments/assets/4a10e86c-2366-41fd-95ae-ad66657ba749)
